addons/controls/src/components/ControlsPanel.tsx

```diff
@@ -0,0 +1,14 @@
+import React, { FC } from 'react';
+import { ArgsTable } from '@storybook/components';
+import { useArgs, useArgTypes, useParameter } from '@storybook/api';
+
+interface ControlsParameters {
+  expanded?: boolean;
+}
+
+export const ControlsPanel: FC = () => {
+  const [args, updateArgs] = useArgs();
+  const rows = useArgTypes();
+  const { expanded } = useParameter<ControlsParameters>('controls', {});
+  return <ArgsTable {...{ compact: !expanded, rows, args, updateArgs }} />;
```

Comment:
maybe for readability sake and debuggability as well extract this to a const so it's easier to edit and also to add a debugger after in case it's necessary?

```jsx
const argsTableProps = { compact: !expanded, rows, args, updateArgs }
return <ArgsTable {...argsTableProps} />;
```

Reply:
```
<ArgsTable compact={!expanded} rows={rows} args={args} updateArgs={updateArgs} />
```
even better